extensions/statistics/js/src/admin/components/StatisticsWidgetDateSelectionModal.tsx

```diff
@@ -0,0 +1,149 @@
+import app from 'flarum/admin/app';
+import ItemList from 'flarum/common/utils/ItemList';
+import generateElementId from 'flarum/admin/utils/generateElementId';
+import Modal, { IInternalModalAttrs } from 'flarum/common/components/Modal';
+
+import Mithril from 'mithril';
+import Button from 'flarum/common/components/Button';
+
+import dayjs from 'dayjs';
+import dayjsUtc from 'dayjs/plugin/utc';
+
+dayjs.extend(dayjsUtc);
+
+export interface IDateSelection {
+  /**
+   * Timestamp (seconds, not ms) for start date
+   */
+  start: number;
+  /**
+   * Timestamp (seconds, not ms) for end date
+   */
+  end: number;
+}
+
+export interface IStatisticsWidgetDateSelectionModalAttrs extends IInternalModalAttrs {
+  onModalSubmit: (dates: IDateSelection) => void;
+  value?: IDateSelection;
+}
+
+interface IStatisticsWidgetDateSelectionModalState {
+  inputs: {
+    startDateVal: string;
+    endDateVal: string;
+  };
+  ids: {
+    startDate: string;
+    endDate: string;
+  };
+}
+
+export default class StatisticsWidgetDateSelectionModal extends Modal<IStatisticsWidgetDateSelectionModalAttrs> {
+  /* @ts-expect-error core typings don't allow us to set the type of the state attr :( */
+  state: IStatisticsWidgetDateSelectionModalState = {
+    inputs: {
+      startDateVal: dayjs().format('YYYY-MM-DD'),
+      endDateVal: dayjs().format('YYYY-MM-DD'),
+    },
+    ids: {
+      startDate: generateElementId(),
+      endDate: generateElementId(),
+    },
+  };
+
+  oninit(vnode: Mithril.Vnode<IStatisticsWidgetDateSelectionModalAttrs, this>) {
+    super.oninit(vnode);
+
+    if (this.attrs.value) {
+      this.state.inputs = {
+        startDateVal: dayjs.utc(this.attrs.value.start * 1000).format('YYYY-MM-DD'),
+        endDateVal: dayjs.utc(this.attrs.value.end * 1000).format('YYYY-MM-DD'),
+      };
+    }
+  }
+
+  className(): string {
+    return 'StatisticsWidgetDateSelectionModal';
```

Comment:
Just a visual nitpick 😇
```suggestion
    return 'StatisticsWidgetDateSelectionModal Modal--small';
```